### PR TITLE
chore: remove node link connection managment

### PR DESCRIPTION
**Pull request by joshuef**
atop #1793 , so blocked until that's in.

This coould go further. But this is a decent reduction already.

We could follow this up removing Link almost entirely and PeerSession thereafter

### CODE REVIEW

**Comment by joshuef** [comment]
bors try

**Comment by joshuef** [comment]
bors try-

**Comment by joshuef** [comment]
bors try

File: less [hunk 1]
```diff
@@ -0,0 +1,121 @@
+    Checking sn_node v0.71.0 (/Users/josh/Projects/sn/sn_node)
```
Comment (oetyng) [inline less:1]: This one slipped in?

**Comment by joshuef** [comment]
bors try

**Comment by joshuef** [comment]
Closing this as it's a failing. There's follow up PRs with various parts of this, and we can pull out more changes as/when those are accepted

**Pull request closed without merging**